### PR TITLE
Fixes file system probing error when parsing empty partitions

### PR DESCRIPTION
**Pull request by hsong-rh**
File system probing will fail if disk has empty partitions in the beginning. Add logic to keep scanning in this scenario.

https://bugzilla.redhat.com/show_bug.cgi?id=1527182

### CODE REVIEW

**Comment by hsong-rh** [comment]
@roliveri @jerryk55 Please review.

File: .codeclimate.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@ engines:
       languages:
         ruby:
           mass_threshold: 25
+          count_threshold: 5
```
Comment (roliveri) [inline .codeclimate.yml]: What does tis control?

Comment (hsong-rh) [reply .codeclimate.yml]: It controls how many similar codes are allowed.

File: lib/fs/MiqFS/FsProbe.rb [hunk 2]
```diff
@@ -25,13 +25,24 @@ def self.getFsMod(dobj, probes = nil)
     partNum = dobj.partNum
 
     probes.each do |pmod|
-      $log.debug "MIQ(FsProbe-getFsMod) FS probe attempting [#{pmod}] for [#{fname}] [partition: #{partNum}]"
+      $log.debug("MIQ(FsProbe-getFsMod) FS probe attempting [#{pmod}] for [#{fname}] [partition: #{partNum}]")
+      unless dobj.kind_of?(MiqDisk)
```
[the remark this answers is not in the document]
Comment (hsong-rh) [reply lib/fs/MiqFS/FsProbe.rb]: This checking to make sure the input parameter is in right type is not related to filesystem probing directly.

File: lib/fs/MiqFS/modules/Ext3Probe.rb [hunk 3]
```diff
@@ -2,24 +2,16 @@
 
 module Ext3Probe
   def self.probe(dobj)
-    $log.debug("Ext3Probe >> dobj=#{dobj}") if $log
```
[the remark this answers is not in the document]
Comment (hsong-rh) [reply lib/fs/MiqFS/modules/Ext3Probe.rb]: This message is surplus. Outside of the probing loop, the similar message has been outputted.

File: lib/fs/MiqFS/FsProbe.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (roliveri) [reply lib/fs/MiqFS/FsProbe.rb]: Yes, it is related to filesystem probing. Different filesystems may be able to handle different types. Actually, we probably shouldn't check types at all, and just rely on duck typing. For now, I see no need to change it.

File: lib/fs/MiqFS/modules/Ext3Probe.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (roliveri) [reply lib/fs/MiqFS/modules/Ext3Probe.rb]: I was referring to the code following the comment.

File: lib/fs/MiqFS/modules/Ext3Probe.rb [hunk 4]
```diff
@@ -2,24 +2,21 @@
...
     unless dobj.kind_of?(MiqDisk)
-      $log.debug "Ext3Probe << FALSE because Disk Object class is not MiqDisk, but is '#{dobj.class}'" if $log
+      $log.debug("Ext3Probe << FALSE because Disk Object class is not MiqDisk, but is '#{dobj.class}'")
```
Comment (roliveri) [inline lib/fs/MiqFS/modules/Ext3Probe.rb]: Should this be:
```ruby
$log&.debug(...
```

File: lib/fs/MiqFS/modules/Ext3Probe.rb [hunk 5]
```diff
@@ -2,24 +2,21 @@
...
-      $log.debug("Ext3Probe << TRUE")
-      return true
-    rescue => err
-      $log.debug "Ext3Probe << FALSE because #{err.message}" if $log
-      return false
-    ensure
-      dobj.seek(0, IO::SEEK_SET)
-    end
+    # If initializing the superblock does not throw any errors, then this is ext3
+    $log.debug("Ext3Probe << TRUE")
```
Comment (roliveri) [inline lib/fs/MiqFS/modules/Ext3Probe.rb]: Should this be:
```ruby
$log&.debug(...
```

File: lib/fs/MiqFS/modules/Ext3Probe.rb [hunk 6]
```diff
@@ -2,24 +2,21 @@
...
+    return true
+  rescue => err
+    $log.debug("Ext3Probe << FALSE because #{err.message}")
```
Comment (roliveri) [inline lib/fs/MiqFS/modules/Ext3Probe.rb]: Should this be:
```ruby
$log&.debug(...
```

File: lib/fs/MiqFS/modules/Ext4Probe.rb [hunk 7]
```diff
@@ -2,24 +2,21 @@
 
 module Ext4Probe
   def self.probe(dobj)
-    $log.debug("Ext4Probe >> dobj=#{dobj}") if $log
     unless dobj.kind_of?(MiqDisk)
-      $log.debug "Ext4Probe << FALSE because Disk Object class is not MiqDisk, but is '#{dobj.class}'" if $log
+      $log.debug("Ext4Probe << FALSE because Disk Object class is not MiqDisk, but is '#{dobj.class}'")
```
Comment (roliveri) [inline lib/fs/MiqFS/modules/Ext4Probe.rb]: Should this be:
```ruby
$log&.debug(...
```

File: lib/fs/MiqFS/modules/Ext4Probe.rb [hunk 8]
```diff
@@ -2,24 +2,21 @@
...
-      $log.debug("Ext4Probe << TRUE")
-      return true
-    rescue => err
-      $log.debug "Ext4Probe << FALSE because #{err.message}" if $log
-      return false
-    ensure
-      dobj.seek(0, IO::SEEK_SET)
-    end
+    # If initializing the superblock does not throw any errors, then this is Ext4
+    $log.debug("Ext4Probe << TRUE")
```
Comment (roliveri) [inline lib/fs/MiqFS/modules/Ext4Probe.rb]: Should this be:
```ruby
$log&.debug(...
```

File: lib/fs/MiqFS/modules/Ext4Probe.rb [hunk 9]
```diff
@@ -2,24 +2,21 @@
...
+    return true
+  rescue => err
+    $log.debug("Ext4Probe << FALSE because #{err.message}")
```
Comment (roliveri) [inline lib/fs/MiqFS/modules/Ext4Probe.rb]: Should this be:
```ruby
$log&.debug(...
```

File: lib/fs/MiqFS/modules/XFSProbe.rb [hunk 10]
```diff
@@ -2,25 +2,22 @@
 
 module XFSProbe
   def self.probe(dobj)
-    $log.debug("XFSProbe >> dobj=#{dobj}") if $log
     unless dobj.kind_of?(MiqDisk)
-      $log.debug "XFSProbe << FALSE because Disk Object class is not MiqDisk, but is '#{dobj.class}'" if $log
+      $log.debug("XFSProbe << FALSE because Disk Object class is not MiqDisk, but is '#{dobj.class}'")
```
Comment (roliveri) [inline lib/fs/MiqFS/modules/XFSProbe.rb]: Should this be:
```ruby
$log&.debug(...
```

File: lib/fs/MiqFS/modules/XFSProbe.rb [hunk 11]
```diff
@@ -2,25 +2,22 @@
...
-      $log.debug("XFSProbe << TRUE")
-      return true
-    rescue => err
-      $log.debug "XFSProbe << FALSE because #{err.message}" if $log
-      return false
-    ensure
-      dobj.seek(0, IO::SEEK_SET)
-    end
+    # If initializing the superblock does not throw any errors, then this is XFS
+    $log.debug("XFSProbe << TRUE")
```
Comment (roliveri) [inline lib/fs/MiqFS/modules/XFSProbe.rb]: Should this be:
```ruby
$log&.debug(...
```

File: lib/fs/MiqFS/modules/XFSProbe.rb [hunk 12]
```diff
@@ -2,25 +2,22 @@
...
+    return true
+  rescue => err
+    $log.debug("XFSProbe << FALSE because #{err.message}")
```
Comment (roliveri) [inline lib/fs/MiqFS/modules/XFSProbe.rb]: Should this be:
```ruby
$log&.debug(...
```

**Comment by hsong-rh** [comment]
@roliveri Changes were made based on your comments. Please review.

**Comment by simaishi** [comment]
Fine backport (to manageiq-gems-pending repo) details:

```
$ git log -1
commit 5bf2750534f8bbe82375b100fc341f1212dd507c
Author: Richard Oliveri <oliveri.richard.github@gmail.com>
Date:   Tue Jan 16 09:58:07 2018 -0500

    Merge pull request #48 from hsong-rh/begins_with_empty_partitions
    
    Fixes file system probing error when parsing empty partitions
    (cherry picked from commit 3943f65c8aa89d6ece3858a67c22e8889964a8ec)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1565362
```

**Comment by simaishi** [comment]
@miq-bot add_label fine/backported
@miq-bot remove_label fine/yes

**Pull request merged**